argocd/apps/templates/namespaces.yaml

```diff
@@ -0,0 +1,14 @@
+{{- range $k, $v := index .Values.namespaces }}
+---
+apiVersion: "v1"
+kind: "Namespace"
+metadata:
+  name: "{{ $k }}"
+  annotations:
+    argocd.argoproj.io/sync-wave: "-99"
+    {{- if $v }}
+    {{- if and (ne $k "kube-prometheus-stack") (ne $k "aad-pod-identity") (ne $k "csi-secrets-store-provider-azure") (ne $k "loki-stack")}}
```

Comment:
Why do you need to exclude these apps specifically?

Reply:
@raphink because they have "deamonset" pods and adding the annotation to their namespaces will force scheduling their pods on node pools. We use affinities for their "deployment/statefulset" pods instead of annotating the namespace.